File: leetcode/30-day-leetcoding-challenge/bitwise_and_of_numbers_range.cpp
class Solution {
public:
    int rangeBitwiseAnd(int m, int n) {
        int count = 0;
        while (m != n) {
            m >>= 1;
            n >>= 1;
            count++;
        }
        return m << count;
    }
};

// https://leetcode.com/problems/bitwise-and-of-numbers-range/

File: leetcode/february-leetcoding-challenge-2024/bitwise_and_of_numbers_range.java
class Solution {
    public int rangeBitwiseAnd(int left, int right) {
        while (left < right) {
            right = right & (right - 1);
        }
        return left & right;
    }
}

// https://leetcode.com/problems/bitwise-and-of-numbers-range/?envType=daily-question&envId=2024-02-21